source/common/upstream/outlier_detection_impl.cc

```diff
@@ -248,6 +248,13 @@ void DetectorImpl::onConsecutive5xxWorker(HostSharedPtr host) {
     return;
   }
 
+  // There are two outcomes here. The ejection will be enforced,
```

Comment:
This comment still doesn't make sense unless I am confused. There is only 1 outcome here. The ejection has been enforced at this point, the host is being ejected, and we are resetting. Please also reflow the comment.

Also, I think the reset should happen after the ejection to avoid a small race condition and re-fire of the event.